src/lfortran/semantics/ast_symboltable_visitor.cpp

```diff
@@ -374,7 +374,109 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
         return r;
     }
 
+    void populate_implicit_dictonary(const AST::Function_t &x, std::map<std::string, ASR::ttype_t*> &implicit_dictionary){
```

Comment:
Let's be consistent:
```suggestion
    void populate_implicit_dictonary(const AST::Function_t &x, std::map<std::string, ASR::ttype_t*> &implicit_dictionary) {
```
and the same below.

Reply:
Apologies for the random drive-by comment, but is "dictonary" supposed to be "dictionary" in the function's name?

Reply:
@digital-carver yes, I think it does, thanks for spotting it!